chore: add return types to build.ts

tamirazrab opened the pull request:
## Description

Add types to mentioned functions in issue. 

## Related Issue

Relates to #1649 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit, [Journey](https://github.com/defenseunicorns/pepr/tree/main/journey), [E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples), [docs](https://github.com/defenseunicorns/pepr/tree/main/docs), [adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or updated as needed
- [x] [Contributor Guide Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request) followed

samayer12 committed:
Encourage function return types

cmwylie19 committed:
Merge branch 'main' into hotfix-return-types-in-funcs

samayer12 committed:
Merge branch 'main' into hotfix-return-types-in-funcs

samayer12 committed:
Merge branch 'main' into hotfix-return-types-in-funcs

cmwylie19 committed:
Merge branch 'main' into hotfix-return-types-in-funcs

samayer12 committed:
Merge branch 'main' into hotfix-return-types-in-funcs

samayer12 committed:
Merge branch 'main' into hotfix-return-types-in-funcs

samayer12 committed:
Merge branch 'main' into hotfix-return-types-in-funcs

samayer12 committed:
Merge branch 'main' into hotfix-return-types-in-funcs

samayer12 committed:
Merge branch 'main' into hotfix-return-types-in-funcs

samayer12 committed:
Merge branch 'main' into hotfix-return-types-in-funcs

samayer12 committed:
Merge branch 'main' into hotfix-return-types-in-funcs

samayer12 committed:
Merge branch 'main' into hotfix-return-types-in-funcs

tamirazrab committed:
chore: add return type for func build.ts

dependabot[bot] committed:
chore: bump github/codeql-action from 3.28.0 to 3.28.1 (#1650)

Bumps [github/codeql-action](https://github.com/github/codeql-action)
from 3.28.0 to 3.28.1.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a
href="https://github.com/github/codeql-action/releases">github/codeql-action's
releases</a>.</em></p>
<blockquote>
<h2>v3.28.1</h2>
<h1>CodeQL Action Changelog</h1>
<p>See the <a
href="https://github.com/github/codeql-action/releases">releases
page</a> for the relevant changes to the CodeQL CLI and language
packs.</p>
<h2>3.28.1 - 10 Jan 2025</h2>
<ul>
<li>CodeQL Action v2 is now deprecated, and is no longer updated or
supported. For better performance, improved security, and new features,
upgrade to v3. For more information, see <a
href="https://github.blog/changelog/2025-01-10-code-scanning-codeql-action-v2-is-now-deprecated/">this
changelog post</a>. <a
href="https://redirect.github.com/github/codeql-action/pull/2677">#2677</a></li>
<li>Update default CodeQL bundle version to 2.20.1. <a
href="https://redirect.github.com/github/codeql-action/pull/2678">#2678</a></li>
</ul>
<p>See the full <a
href="https://github.com/github/codeql-action/blob/v3.28.1/CHANGELOG.md">CHANGELOG.md</a>
for more information.</p>
</blockquote>
</details>
<details>
<summary>Changelog</summary>
<p><em>Sourced from <a
href="https://github.com/github/codeql-action/blob/main/CHANGELOG.md">github/codeql-action's
changelog</a>.</em></p>
<blockquote>
<h1>CodeQL Action Changelog</h1>
<p>See the <a
href="https://github.com/github/codeql-action/releases">releases
page</a> for the relevant changes to the CodeQL CLI and language
packs.</p>
<h2>[UNRELEASED]</h2>
<p>No user facing changes.</p>
<h2>3.28.1 - 10 Jan 2025</h2>
<ul>
<li>CodeQL Action v2 is now deprecated, and is no longer updated or
supported. For better performance, improved security, and new features,
upgrade to v3. For more information, see <a
href="https://github.blog/changelog/2025-01-10-code-scanning-codeql-action-v2-is-now-deprecated/">this
changelog post</a>. <a
href="https://redirect.github.com/github/codeql-action/pull/2677">#2677</a></li>
<li>Update default CodeQL bundle version to 2.20.1. <a
href="https://redirect.github.com/github/codeql-action/pull/2678">#2678</a></li>
</ul>
<h2>3.28.0 - 20 Dec 2024</h2>
<ul>
<li>Bump the minimum CodeQL bundle version to 2.15.5. <a
href="https://redirect.github.com/github/codeql-action/pull/2655">#2655</a></li>
<li>Don't fail in the unusual case that a file is on the search path. <a
href="https://redirect.github.com/github/codeql-action/pull/2660">#2660</a>.</li>
</ul>
<h2>3.27.9 - 12 Dec 2024</h2>
<p>No user facing changes.</p>
<h2>3.27.8 - 12 Dec 2024</h2>
<ul>
<li>Fixed an issue where streaming the download and extraction of the
CodeQL bundle did not respect proxy settings. <a
href="https://redirect.github.com/github/codeql-action/pull/2624">#2624</a></li>
</ul>
<h2>3.27.7 - 10 Dec 2024</h2>
<ul>
<li>We are rolling out a change in December 2024 that will extract the
CodeQL bundle directly to the toolcache to improve performance. <a
href="https://redirect.github.com/github/codeql-action/pull/2631">#2631</a></li>
<li>Update default CodeQL bundle version to 2.20.0. <a
href="https://redirect.github.com/github/codeql-action/pull/2636">#2636</a></li>
</ul>
<h2>3.27.6 - 03 Dec 2024</h2>
<ul>
<li>Update default CodeQL bundle version to 2.19.4. <a
href="https://redirect.github.com/github/codeql-action/pull/2626">#2626</a></li>
</ul>
<h2>3.27.5 - 19 Nov 2024</h2>
<p>No user facing changes.</p>
<h2>3.27.4 - 14 Nov 2024</h2>
<p>No user facing changes.</p>
<h2>3.27.3 - 12 Nov 2024</h2>
<p>No user facing changes.</p>
<h2>3.27.2 - 12 Nov 2024</h2>
<ul>
<li>Fixed an issue where setting up the CodeQL tools would sometimes
fail with the message &quot;Invalid value 'undefined' for header
'authorization'&quot;. <a
href="https://redirect.github.com/github/codeql-action/pull/2590">#2590</a></li>
</ul>
<!-- raw HTML omitted -->
</blockquote>
<p>... (truncated)</p>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/github/codeql-action/commit/b6a472f63d85b9c78a3ac5e89422239fc15e9b3c"><code>b6a472f</code></a>
Merge pull request <a
href="https://redirect.github.com/github/codeql-action/issues/2681">#2681</a>
from github/update-v3.28.1-ea6acbfea</li>
<li><a
href="https://github.com/github/codeql-action/commit/bb999b434f581db70696c32bf01941ec0391c3cb"><code>bb999b4</code></a>
Update changelog for v3.28.1</li>
<li><a
href="https://github.com/github/codeql-action/commit/ea6acbfeae40310725526bdaea16fb48c1f62be0"><code>ea6acbf</code></a>
Merge pull request <a
href="https://redirect.github.com/github/codeql-action/issues/2677">#2677</a>
from github/angelapwen/deprecate-action-v2</li>
<li><a
href="https://github.com/github/codeql-action/commit/4df151edec34bbfb2f213ed8b2dc74363baa69ca"><code>4df151e</code></a>
Merge branch 'main' into angelapwen/deprecate-action-v2</li>
<li><a
href="https://github.com/github/codeql-action/commit/a05a7eb09cb4faebee87507a854cbd429fbe6bc6"><code>a05a7eb</code></a>
Fix PR number in changenote</li>
<li><a
href="https://github.com/github/codeql-action/commit/8d2753b250830f4073d10fc13a3264a17ba82a20"><code>8d2753b</code></a>
Add public changelog blog post link</li>
<li><a
href="https://github.com/github/codeql-action/commit/e83e0a4f58f2ca25f7dd222e8689519a74bf26fc"><code>e83e0a4</code></a>
Merge pull request <a
href="https://redirect.github.com/github/codeql-action/issues/2673">#2673</a>
from github/dependabot/npm_and_yarn/npm-877f465710</li>
<li><a
href="https://github.com/github/codeql-action/commit/b7ff30899f3f4aa6705d70c1456d7189e1b320ba"><code>b7ff308</code></a>
Merge pull request <a
href="https://redirect.github.com/github/codeql-action/issues/2678">#2678</a>
from github/update-bundle/codeql-bundle-v2.20.1</li>
<li><a
href="https://github.com/github/codeql-action/commit/1aa16c2c36f41e9ce531bdcd2aa834171700e48d"><code>1aa16c2</code></a>
Merge branch 'main' into update-bundle/codeql-bundle-v2.20.1</li>
<li><a
href="https://github.com/github/codeql-action/commit/fb65b6ce7884900fde5b15518bec92ad6875180e"><code>fb65b6c</code></a>
Merge pull request <a
href="https://redirect.github.com/github/codeql-action/issues/2672">#2672</a>
from github/mbg/start-proxy/include-type-in-urls-output</li>
<li>Additional commits viewable in <a
href="https://github.com/github/codeql-action/compare/48ab28a6f5dbc2a99bf1e0131198dd8f1df78169...b6a472f63d85b9c78a3ac5e89422239fc15e9b3c">compare
view</a></li>
</ul>
</details>
<br />


[![Dependabot compatibility
score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=github/codeql-action&package-manager=github_actions&previous-version=3.28.0&new-version=3.28.1)](https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores)

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)


</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

cmwylie19 commented on .eslintrc.json:
```diff
@@ -21,6 +21,15 @@
+        "@typescript-eslint/explicit-function-return-type": "warn"
+      }
+    }
+  ],
```
```suggestion
```

Although we are using this config to check to ensure we have finished return types for certain files, we do not want to turn this on yet. We have a meeting for TypeScript standards where we will talk about this.

samayer12 replied:
This change is part of #1364 and should not be included in this PR. Good catch, @cmwylie19

tamirazrab replied:
Yes, I branched off from #1364 thought it might serve as the starting point for this issue.

samayer12 replied:
Agreed, I do the same thing locally and exclude `.eslintrc.json` from my commits until #1364 loses the `ON HOLD` label and is merged.

samayer12 replied:
#1364 has been merged.

cmwylie19 commented on src/cli/build.ts:
```diff
@@ -120,6 +149,9 @@ export default function (program: RootCmd): void {
             description: cfg.description,
+            alwaysIgnore: {
+              namespaces: []
+            },
```
```suggestion
```

setting `alwaysIgnore.namespaces[]` here would override the config placed in the `pepr` section of `package.json`. This needs to come from cfg.pepr.

samayer12 replied:
Fixed in [65bcfc4](https://github.com/defenseunicorns/pepr/pull/1651/commits/65bcfc432198be7227cf87a9b0d3edcccfba5e89)

cmwylie19 commented on src/cli/build.ts:
```diff
@@ -23,11 +23,40 @@ import {
+};
+
+type LoadModuleReturn = {
+  cfg: PeprConfig;
```
```suggestion
  cfg: PeprConfig;
```

Pretty sure this is not not technically accurate, these are the [Pepr Config docs](https://docs.pepr.dev/main/user-guide/customization/#packagejson-configurations-table)

samayer12 replied:
Reviewed pepr docs against the implementation, it looks like we cover the relevant properties.

tamirazrab committed:
fix: override alwaysIgnore config & include all fields from module config in pepr config

samayer12 committed:
chore: add return types to loader.ts (#1648)

## Description

In support of #1364, this PR adds typing to previously-untyped
functions.

End to End Test:  <!-- if applicable -->  
(See [Pepr Excellent
Examples](https://github.com/defenseunicorns/pepr-excellent-examples))

## Related Issue

Fixes #
<!-- or -->
Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [ ] Unit,
[Journey](https://github.com/defenseunicorns/pepr/tree/main/journey),
[E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples),
[docs](https://github.com/defenseunicorns/pepr/tree/main/docs),
[adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or
updated as needed
- [ ] [Contributor Guide
Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request)
followed

dependabot[bot] committed:
chore: bump @types/node from 22.10.5 to 22.10.6 in the development-dependencies group (#1653)

Bumps the development-dependencies group with 1 update:
[@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node).

Updates `@types/node` from 22.10.5 to 22.10.6
<details>
<summary>Commits</summary>
<ul>
<li>See full diff in <a
href="https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node">compare
view</a></li>
</ul>
</details>
<br />

<details>
<summary>Most Recent Ignore Conditions Applied to This Pull
Request</summary>

| Dependency Name | Ignore Conditions |
| --- | --- |
| @types/node | [>= 20.a, < 21] |
</details>


[![Dependabot compatibility
score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=@types/node&package-manager=npm_and_yarn&previous-version=22.10.5&new-version=22.10.6)](https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores)

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore <dependency name> major version` will close this
group update PR and stop Dependabot creating any more for the specific
dependency's major version (unless you unignore this specific
dependency's major version or upgrade to it yourself)
- `@dependabot ignore <dependency name> minor version` will close this
group update PR and stop Dependabot creating any more for the specific
dependency's minor version (unless you unignore this specific
dependency's minor version or upgrade to it yourself)
- `@dependabot ignore <dependency name>` will close this group update PR
and stop Dependabot creating any more for the specific dependency
(unless you unignore this specific dependency or upgrade to it yourself)
- `@dependabot unignore <dependency name>` will remove all of the ignore
conditions of the specified dependency
- `@dependabot unignore <dependency name> <ignore condition>` will
remove the ignore condition of the specified dependency and ignore
conditions


</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Sam Mayer <sam.mayer@defenseunicorns.com>

samayer12 committed:
chore: remove unused property in assets class (#1647)

## Description

This PR removes the `Assets.hash` property and calculates the hash of a
pepr module _just_ before we need to provide that value to a consuming
function. This will avoid potential issues with state and removes some
dead code.


End to End Test:  <!-- if applicable -->  
(See [Pepr Excellent
Examples](https://github.com/defenseunicorns/pepr-excellent-examples))

## Related Issue

Fixes #1638 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging
- [ ] Unit,
[Journey](https://github.com/defenseunicorns/pepr/tree/main/journey),
[E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples),
[docs](https://github.com/defenseunicorns/pepr/tree/main/docs),
[adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or
updated as needed
- [ ] [Contributor Guide
Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request)
followed

Co-authored-by: Barrett <81570928+btlghrants@users.noreply.github.com>

samayer12 committed:
chore: encourage return types in function definitions (#1364)

## Description

Encourage non-test code to specify return types.

## Related Issue

Hotifx

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit,
[Journey](https://github.com/defenseunicorns/pepr/tree/main/journey),
[E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples),
[docs](https://github.com/defenseunicorns/pepr/tree/main/docs),
[adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or
updated as needed
- [x] [Contributor Guide
Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request)
followed

Co-authored-by: Case Wylie <cmwylie19@defenseunicorns.com>

samayer12 committed:
fix(CI): grant write permission to CI job (#1658)

## Description

A CI job fails due to what may be a permissions issue. This PR adds
permissions.

End to End Test:  <!-- if applicable -->  
(See [Pepr Excellent
Examples](https://github.com/defenseunicorns/pepr-excellent-examples))

## Related Issue

Fixes #1657 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit,
[Journey](https://github.com/defenseunicorns/pepr/tree/main/journey),
[E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples),
[docs](https://github.com/defenseunicorns/pepr/tree/main/docs),
[adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or
updated as needed
- [x] [Contributor Guide
Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request)
followed

cmwylie19 committed:
feat: additionalIgnoredNamespaces at runtime through helm (#1641)

## Description

This feature:
- [x] Adds additionalIgnoredNamespaces to values.yaml
- [x] Sets additionalIgnoredNamespaces in controller container envs as
`PEPR_ADDITIONAL_IGNORED_NAMESPACES`
- [x] reads `PEPR_ADDITIONAL_IGNORED_NAMESPACES` env and adds namespaces
to ignoredNamespaces
- [x] Does **not** document PEPR_ADDITIONAL_IGNORED_NAMESPACES as
feature is meant to be set through helm chart and not by hand, By hand
you should set them in `package.json`

In action:
`package.json`

```json
    "alwaysIgnore": {
      "namespaces": ["something"]
    },
```


`values.yaml`
```yaml
additionalIgnoredNamespaces: 
  - 'kube-system'
  - 'kube-public'
  - 'kube-node-lease'
  - 'default'
  - 'pepr'
  - 'pepr-system'
  - 'pepr-test-module'
```

`> helm template .`

`controllers`
```yaml
            - name: PEPR__ADDITIONAL_IGNORED_NAMESPACES
              value: "kube-system, kube-public, kube-node-lease, default, pepr, pepr-system, pepr-test-module"
```

`webhook configs`

```yaml
    namespaceSelector:
      matchExpressions:
        - key: kubernetes.io/metadata.name
          operator: NotIn
          values:
            - kube-system
            - pepr-system
            - kube-system
            - kube-public
            - kube-node-lease
            - default
            - pepr
            - pepr-system
            - pepr-test-module
            - something
```

## Related Issue

Fixes #1610 #1617 
<!-- or -->
Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit,
[Journey](https://github.com/defenseunicorns/pepr/tree/main/journey),
[E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples),
[docs](https://github.com/defenseunicorns/pepr/tree/main/docs),
[adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or
updated as needed
- [x] [Contributor Guide
Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request)
followed

---------

Signed-off-by: Case Wylie <cmwylie19@defenseunicorns.com>

samayer12 committed:
chore: remove unused mocking (#1664)

## Description

I noticed that `__mocks__` is an old and unused directory, let's take it
out.

End to End Test:  <!-- if applicable -->  
(See [Pepr Excellent
Examples](https://github.com/defenseunicorns/pepr-excellent-examples))

## Related Issue

None

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit,
[Journey](https://github.com/defenseunicorns/pepr/tree/main/journey),
[E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples),
[docs](https://github.com/defenseunicorns/pepr/tree/main/docs),
[adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or
updated as needed
- [x] [Contributor Guide
Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request)
followed

samayer12 committed:
chore: remove circular dependencies in `src/lib/assets/` (#1652)

## Description

This PR removes circular dependencies.

End to End Test:  <!-- if applicable -->  
(See [Pepr Excellent
Examples](https://github.com/defenseunicorns/pepr-excellent-examples))

## Related Issue

Fixes #1632 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit,
[Journey](https://github.com/defenseunicorns/pepr/tree/main/journey),
[E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples),
[docs](https://github.com/defenseunicorns/pepr/tree/main/docs),
[adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or
updated as needed
- [x] [Contributor Guide
Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request)
followed

---------

Co-authored-by: Case Wylie <cmwylie19@defenseunicorns.com>

btlghrants committed:
test: add cli test to validate multiple manifest generation paths result in comparable output (#1642)

## Description

Adds a new `npm run test:int` (int == "integration") script to invoke a
new style of test -- integration tests -- where "integration" means
something like "wider in scope than unit tests" but "not requiring a
full K8s cluster" (general idea is to have a nice place for
non-k8s-required tests to land (e.g. client-side `pepr` CLI commands,
like init / build).

## Related Issue

Fixes #1654 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit,
[Journey](https://github.com/defenseunicorns/pepr/tree/main/journey),
[E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples),
[docs](https://github.com/defenseunicorns/pepr/tree/main/docs),
[adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or
updated as needed
- [x] [Contributor Guide
Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request)
followed

---------

Co-authored-by: Sam Mayer <sam.mayer@defenseunicorns.com>

samayer12 committed:
chore: remove max-statements warning in `schedule.test.ts` (#1671)

## Description

This PR refactors a large test to use parameterized testing.

## Related Issue

Relates to #1247 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit,
[Journey](https://github.com/defenseunicorns/pepr/tree/main/journey),
[E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples),
[docs](https://github.com/defenseunicorns/pepr/tree/main/docs),
[adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or
updated as needed
- [x] [Contributor Guide
Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request)
followed

dependabot[bot] committed:
chore: bump undici from 7.2.1 to 7.2.2 in the development-dependencies group (#1673)

Bumps the development-dependencies group with 1 update:
[undici](https://github.com/nodejs/undici).

Updates `undici` from 7.2.1 to 7.2.2
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a
href="https://github.com/nodejs/undici/releases">undici's
releases</a>.</em></p>
<blockquote>
<h2>v7.2.2</h2>
<h2>What's Changed</h2>
<ul>
<li>Update WPT by <a
href="https://github.com/github-actions"><code>@​github-actions</code></a>
in <a
href="https://redirect.github.com/nodejs/undici/pull/3991">nodejs/undici#3991</a></li>
<li>fix: dual-stack retries infinite loop by <a
href="https://github.com/luddd3"><code>@​luddd3</code></a> in <a
href="https://redirect.github.com/nodejs/undici/pull/4001">nodejs/undici#4001</a></li>
<li>build(deps): bump codecov/codecov-action from 5.0.7 to 5.1.2 by <a
href="https://github.com/dependabot"><code>@​dependabot</code></a> in <a
href="https://redirect.github.com/nodejs/undici/pull/3980">nodejs/undici#3980</a></li>
<li>build(deps): bump github/codeql-action from 3.27.0 to 3.28.1 by <a
href="https://github.com/dependabot"><code>@​dependabot</code></a> in <a
href="https://redirect.github.com/nodejs/undici/pull/4003">nodejs/undici#4003</a></li>
<li>tests: Update Cache Tests by <a
href="https://github.com/github-actions"><code>@​github-actions</code></a>
in <a
href="https://redirect.github.com/nodejs/undici/pull/3965">nodejs/undici#3965</a></li>
<li>ReadableStreamFrom pull until cannot on empty enqueu by <a
href="https://github.com/KhafraDev"><code>@​KhafraDev</code></a> in <a
href="https://redirect.github.com/nodejs/undici/pull/4002">nodejs/undici#4002</a></li>
<li>Resolve circular dependency by <a
href="https://github.com/mcollina"><code>@​mcollina</code></a> in <a
href="https://redirect.github.com/nodejs/undici/pull/4006">nodejs/undici#4006</a></li>
</ul>
<p><strong>Full Changelog</strong>: <a
href="https://github.com/nodejs/undici/compare/v7.2.1...v7.2.2">https://github.com/nodejs/undici/compare/v7.2.1...v7.2.2</a></p>
</blockquote>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/nodejs/undici/commit/bd98a6303e45d5e0d44192a93731b1defdb415f3"><code>bd98a63</code></a>
Bumped v7.2.2 (<a
href="https://redirect.github.com/nodejs/undici/issues/4007">#4007</a>)</li>
<li><a
href="https://github.com/nodejs/undici/commit/e8a93e59246453e359ff02375d7111f8c10d4e40"><code>e8a93e5</code></a>
Resolve circular dependency (<a
href="https://redirect.github.com/nodejs/undici/issues/4006">#4006</a>)</li>
<li><a
href="https://github.com/nodejs/undici/commit/b55476909055bf698cd8eeabe081daa1cc6d9d4c"><code>b554769</code></a>
Revert &quot;chore: update cache tests (<a
href="https://redirect.github.com/nodejs/undici/issues/3965">#3965</a>)&quot;</li>
<li><a
href="https://github.com/nodejs/undici/commit/90a5d60a8576e974e4122bf694e12ec7e6d0d20f"><code>90a5d60</code></a>
ReadableStreamFrom pull until cannot on empty enqueu (<a
href="https://redirect.github.com/nodejs/undici/issues/4002">#4002</a>)</li>
<li><a
href="https://github.com/nodejs/undici/commit/09077922232ce28fd0528f27bf3d7316d6a78c91"><code>0907792</code></a>
chore: update cache tests (<a
href="https://redirect.github.com/nodejs/undici/issues/3965">#3965</a>)</li>
<li><a
href="https://github.com/nodejs/undici/commit/e3928edfc3c121e170ee6068f36e4c878ee09b14"><code>e3928ed</code></a>
build(deps): bump github/codeql-action from 3.27.0 to 3.28.1 (<a
href="https://redirect.github.com/nodejs/undici/issues/4003">#4003</a>)</li>
<li><a
href="https://github.com/nodejs/undici/commit/c4ee20f8d0f565f91621ad20088b44e39372fcd0"><code>c4ee20f</code></a>
build(deps): bump codecov/codecov-action from 5.0.7 to 5.1.2 (<a
href="https://redirect.github.com/nodejs/undici/issues/3980">#3980</a>)</li>
<li><a
href="https://github.com/nodejs/undici/commit/fd01347b6e2dc2163033dc8db33a61c0d6157d80"><code>fd01347</code></a>
fix: dual-stack retries infinite loop (<a
href="https://redirect.github.com/nodejs/undici/issues/4001">#4001</a>)</li>
<li><a
href="https://github.com/nodejs/undici/commit/45eaed232577e9b66d2f0a596eb018c79e098541"><code>45eaed2</code></a>
chore: update WPT (<a
href="https://redirect.github.com/nodejs/undici/issues/3991">#3991</a>)</li>
<li>See full diff in <a
href="https://github.com/nodejs/undici/compare/v7.2.1...v7.2.2">compare
view</a></li>
</ul>
</details>
<br />


[![Dependabot compatibility
score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=undici&package-manager=npm_and_yarn&previous-version=7.2.1&new-version=7.2.2)](https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores)

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore <dependency name> major version` will close this
group update PR and stop Dependabot creating any more for the specific
dependency's major version (unless you unignore this specific
dependency's major version or upgrade to it yourself)
- `@dependabot ignore <dependency name> minor version` will close this
group update PR and stop Dependabot creating any more for the specific
dependency's minor version (unless you unignore this specific
dependency's minor version or upgrade to it yourself)
- `@dependabot ignore <dependency name>` will close this group update PR
and stop Dependabot creating any more for the specific dependency
(unless you unignore this specific dependency or upgrade to it yourself)
- `@dependabot unignore <dependency name>` will remove all of the ignore
conditions of the specified dependency
- `@dependabot unignore <dependency name> <ignore condition>` will
remove the ignore condition of the specified dependency and ignore
conditions


</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Case Wylie <cmwylie19@defenseunicorns.com>

cmwylie19 committed:
chore: pass imagePullSecret to helm (#1670)

## Description

Pass build arg `withPullSecret` to helm 

## Related Issue

Fixes #1660 
<!-- or -->
Relates to #1640 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit,
[Journey](https://github.com/defenseunicorns/pepr/tree/main/journey),
[E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples),
[docs](https://github.com/defenseunicorns/pepr/tree/main/docs),
[adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or
updated as needed
- [x] [Contributor Guide
Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request)
followed

---------

Signed-off-by: Case Wylie <cmwylie19@defenseunicorns.com>
Co-authored-by: Barrett LaFrance <barrett.t.lafrance@gmail.com>

cmwylie19 committed:
chore: remove write perms image digest workflow (#1677)

## Description

Remove excess perms from Image Digest Workflow 
## Related Issue

Fixes #1675 
<!-- or -->
Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging
- [x] Unit,
[Journey](https://github.com/defenseunicorns/pepr/tree/main/journey),
[E2E Tests](https://github.com/defenseunicorns/pepr-excellent-examples),
[docs](https://github.com/defenseunicorns/pepr/tree/main/docs),
[adr](https://github.com/defenseunicorns/pepr/tree/main/adr) added or
updated as needed
- [x] [Contributor Guide
Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request)
followed

Signed-off-by: Case Wylie <cmwylie19@defenseunicorns.com>

samayer12 committed:
Apply linter formatting

samayer12 reviewed (dismissed):
This PR resolves the typing warnings and we can use it as a starting point for refactors to our current typing structure.

samayer12 commented:
@tamirazrab, thanks for the contribution! We need to sign the commits on this PR before we can merge it in due to organizational policy. Here are the [GitHub docs on commit signing](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and a post that covers [how to sign existing commits on a branch](https://superuser.com/questions/397149/can-you-gpg-sign-old-commits).

tamirazrab commented:
> @tamirazrab, thanks for the contribution! We need to sign the commits on this PR before we can merge it in due to organizational policy. Here are the [GitHub docs on commit signing](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and a post that covers [how to sign existing commits on a branch](https://superuser.com/questions/397149/can-you-gpg-sign-old-commits).

Thanks for links, I think the results are messed up, it didn't verified. Maybe I could make new branch off main and make new PR with my changes, what do you think?

cmwylie19 commented:
> > @tamirazrab, thanks for the contribution! We need to sign the commits on this PR before we can merge it in due to organizational policy. Here are the [GitHub docs on commit signing](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and a post that covers [how to sign existing commits on a branch](https://superuser.com/questions/397149/can-you-gpg-sign-old-commits).
> 
> Thanks for links, I think the results are messed up, it didn't verified. Maybe I could make new branch off main and make new PR with my changes, what do you think?

Yes @tamirazrab , that will work.

tamirazrab committed:
chore(test): sign commit

tamirazrab commented:
Closed in favor of #1687
